service/backend/pchain/construction.go

```diff
@@ -45,7 +48,25 @@ func (*Backend) ConstructionPreprocess(
 	if reqMetadata == nil {
 		reqMetadata = make(map[string]interface{})
 	}
+
+	// Build a dummy base tx to calculate the base fee
+	dummyBaseTx, _, err := pmapper.BuildTx(
+		pmapper.OpDummyBase,
+		matches,
+		pmapper.Metadata{BlockchainID: ids.Empty},
+		b.codec,
+		b.avaxAssetID,
+	)
+	if err != nil {
+		return nil, service.WrapError(service.ErrInternalError, err)
+	}
+	fee, err := b.calculateFee(ctx, dummyBaseTx, 0)
```

Comment:
Preprocessing is an offline endpoint, you cannot talk to the node here. You can only do that in Metadata and Submit. 

I'd recommend passing matches in options and calculating the fee there. This way you should be able to implement this once instead of splitting between import/export and staking cases.

Reply:
Done